lib/matplotlib/tests/test_text.py

```diff
@@ -756,3 +756,17 @@ def test_pdf_chars_beyond_bmp():
     plt.rcParams['mathtext.fontset'] = 'stixsans'
     plt.figure()
     plt.figtext(0.1, 0.5, "Mass $m$ \U00010308", size=30)
+
+
+def test_cache_large_labels():
+    """Test to verify cache helps when ticks are too large"""
+    times = []
+    for pow in range(1, 5):
+        labels = [i for i in range(10**pow)]
+        t0 = time.perf_counter()
+        plt.figure()
```

Comment:
I think you want to create the figure _outside_ of this loop so that we can re-use the `Text` instances that are created.  If you make a new figure every time we also re-create the `Text` instances every time which means we do not get to benefit from the cache.  I think where the saving will come in is if we `fig.draw_without_render` (or save to a BytesIO) the same `Figure` multiple times we will see speed ups on the later ones.